src/services/shims.ts

```diff
@@ -322,8 +323,19 @@ module ts {
         }
     }
 
+    function serializeResultAsJson(target: Shim, key: string | symbol, descriptor: PropertyDescriptor) {

```

Comment:
Oh I think I see what you are saying. Instead of a generic signature, I think I'd prefer to have two PropertyDecorators: StringPropertyDecorator and SymbolPropertyDecorator. And you pick which one you want based on the property being decorated.